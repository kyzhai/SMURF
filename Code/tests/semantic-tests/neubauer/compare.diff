76a77,78
> 	ID: head h : t :: [a] -> a
> 	ID: tail h : t :: [a] -> [a]
