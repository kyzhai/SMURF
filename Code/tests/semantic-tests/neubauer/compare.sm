a = 1 < 2

b = 1 > 2

c = 1 <= 2

d = 1 >= 2

e = 1 $< 2
f = 1. $< 2

g = 1 $> 2
h = 1 $> 2.

i = 1 $<= 2
j = 1 $<= 2.

k = 1 $>= 2
l = 1. $>= 2

m = 4 == 4
n = 4 == 3
main = []
