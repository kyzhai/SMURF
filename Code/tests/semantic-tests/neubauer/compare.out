Program: 
	SVardef: 
		ID: a :: Num

	1 < 2
	SVardef: 
		ID: b :: Num

	1 > 2
	SVardef: 
		ID: c :: Num

	1 <= 2
	SVardef: 
		ID: d :: Num

	1 >= 2
	SVardef: 
		ID: e :: Num

	1 $< 2
	SVardef: 
		ID: f :: Num

	1. $< 2
	SVardef: 
		ID: g :: Num

	1 $> 2
	SVardef: 
		ID: h :: Num

	1 $> 2.
	SVardef: 
		ID: i :: Num

	1 $<= 2
	SVardef: 
		ID: j :: Num

	1 $<= 2.
	SVardef: 
		ID: k :: Num

	1 $>= 2
	SVardef: 
		ID: l :: Num

	1. $>= 2
	SVardef: 
		ID: m :: Num

	4 == 4
	SVardef: 
		ID: n :: Num

	4 == 3
Global Scope: 
	ID: a :: Num
	ID: b :: Num
	ID: c :: Num
	ID: d :: Num
	ID: e :: Num
	ID: f :: Num
	ID: g :: Num
	ID: h :: Num
	ID: i :: Num
	ID: j :: Num
	ID: k :: Num
	ID: l :: Num
	ID: m :: Num
	ID: n :: Num
	ID: print :: Unknown
	ID: random :: Unknown

